src/Docfx.MarkdigEngine.Extensions/PlantUml/FormatterFactory.cs

```diff
@@ -0,0 +1,41 @@
+using PlantUml.Net;
+
+namespace Docfx.MarkdigEngine.Extensions;
+
+internal class FormatterFactory
+{
+    private readonly DocfxPlantUmlSettings settings;
+
+    private OutputFormat OutputFormat => settings.OutputFormat;
+
+    public FormatterFactory(DocfxPlantUmlSettings settings)
+    {
+        this.settings = settings;
+    }
+
+    internal IOutputFormatter CreateOutputFormatter()
+    {
+        switch (OutputFormat)
+        {
+            case OutputFormat.Svg:
+                return new SvgOutputFormatter();
```

Comment:
nit: seems like an overkill to extract the `IOutputFormatter` interface, could this be turned into a simple function in `CustomCodeBlockRenderer`?